### PR TITLE
Add a box for repochanges in the pull request template.

### PR DESCRIPTION
**Pull request by yardasol**
## Summary of changes

This PR adds a checkbox for repo-related changes to the pull request template, a related subbox, and moves the checkbox for adding content to release notes to be contingent on the changes requiring documentation changes

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Required for Merging
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] My change is a source code change
  - [ ] I have added/modified tests to cover my changes
  - [ ] I have explained why my change does not require new/modified tests 
- [x] All new and existing tests passed.
  - [x] CI tests pass
  - [x] Local tests pass (including Serpent2 integration tests)
- [ ] I have recorded my changes in the changelog for the upcoming release

## Associated Issues and PRs

- Issue: none

## Associated Developers

- Dev: @munkm 

## Checklist for Reviewers

Reviewers should use [this link](https://arfc.github.io/manual/guides/pull_requests) to get to the 
Review Checklist before they begin their review.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -15,13 +15,15 @@
 - [ ] My code follows the code style of this project.
 - [ ] My change requires a change to the documentation.
   - [ ] I have updated the documentation accordingly.
+  - [ ] I have recorded my changes in the changelog for the upcoming release
```
Comment (munkm) [inline .github/pull_request_template.md:18]: Do we want to recommend when a change is necessary for the changelog? I do think putting this as a documentation requirement is good, but I'm trying to think about minimal changes that might not be necessary to put in the changelog. Maybe it's sufficient to trust our contributors to make the right call. What if we said something like: 

```suggestion
  - [ ] If I have made a user-facing change, I have recorded my changes in the changelog for the upcoming release 
```

I'm not sure. What do you think @yardasol ?

**Review by munkm — COMMENTED** [review]
Thanks for these updates @yardasol ! I think these are great enhancements for the PR template. I have two questions about this change that I've added as inline comments.

File: .github/pull_request_template.md (hunk 1 above)
Comment (yardasol) [reply .github/pull_request_template.md:18]: Ooh I like that better!

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -15,13 +15,15 @@
 - [ ] My code follows the code style of this project.
 - [ ] My change requires a change to the documentation.
   - [ ] I have updated the documentation accordingly.
+  - [ ] I have recorded my changes in the changelog for the upcoming release 
 - [ ] My change is a source code change
   - [ ] I have added/modified tests to cover my changes
-  - [ ] I have explained why my change does not require new/modified tests 
+  - [ ] I have explained why my change does not require new/modified tests
+- [ ] My change is related to the repository
```
[the remark this answers is not in the document]
Comment (yardasol) [reply .github/pull_request_template.md:22]: I mean things like github actions scripts, pr and issue templates, etc.

**Comment by yardasol** [comment]
@munkm I've made some changes that I think clarify on the points you brought up

**Review by munkm — APPROVED** [review]

**Pull request merged**